src/functions/routers/types.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  FunctionContext,
+  FunctionDefinitionArgs,
+  FunctionParameters,
+  FunctionRuntimeParameters,
+} from "../types.ts";
+
+export type BlockActionHandler<Definition> = Definition extends
+  FunctionDefinitionArgs<infer I, infer O, infer RI, infer RO> ? {
+    (
+      context: ActionContext<FunctionRuntimeParameters<I, RI>>,
+    ): Promise<void> | void;
+  }
+  : never;
+
+export type ActionContext<InputParameters> =
+  & Omit<FunctionContext<InputParameters>, "event">
+  & ActionSpecificContext<InputParameters>;
+
+type ActionSpecificContext<InputParameters extends FunctionParameters> = {
+  body: BlockActionInvocationBody<InputParameters>;
+  action: BlockAction;
+};
+
+// TODO: all below stolen from deno-slack-runtime, need to centralize these somewhere
+export type BlockActionInvocationBody<
+  InputParameters extends FunctionParameters,
+> = {
+  type: string;
+  actions: BlockAction[];
```

Comment:
❓ This is an array of Block Actions being passed to an individual action handler. Do you know what this will actually look like? Is it just all possible Block Actions for the fn?